pkg/controller/usersignup/masteruserrecord_test.go

```diff
@@ -109,10 +109,27 @@ func TestMigrateMurIfNecessary(t *testing.T) {
 		// given
 		nSTemplateTier := newNsTemplateTier("advanced", "654321b", "dev", "stage", "extra")
 		mur := newMasterUserRecord(nSTemplateTier, "johny", operatorNamespace, test.MemberClusterName, "123456789")
-		for index := range mur.Spec.UserAccounts[0].Spec.NSTemplateSet.Namespaces {
-			mur.Spec.UserAccounts[0].Spec.NSTemplateSet.Namespaces[index].TemplateRef = ""
+		mur.Spec.UserAccounts[0].Spec.NSTemplateSet.Namespaces = []v1alpha1.NSTemplateSetNamespace{
+			{
+				Type:        "dev",
+				Revision:    "123abc1",
+				TemplateRef: "",
+			},
+			{
+				Type:        "stage",
+				Revision:    "123abc2",
+				TemplateRef: "",
+			},
+			{
+				Type:        "extra",
+				Revision:    "123abc3",
+				TemplateRef: "",
+			},
+		}
+		mur.Spec.UserAccounts[0].Spec.NSTemplateSet.ClusterResources = &v1alpha1.NSTemplateSetClusterResources{
+			Revision:    "654321b",
+			TemplateRef: "",
 		}
-		mur.Spec.UserAccounts[0].Spec.NSTemplateSet.ClusterResources.TemplateRef = ""
```

Comment:
I thought it would be more explicit/easier to understand that in this particular test, we override the default values from the `mur` to satisfy the prerequisites.